### PR TITLE
[ADF-696] Entire accordion group header should be clickable

### PR DESCRIPTION
**Pull request by monsterflaw**
**Please check if the PR fulfills these requirements**
```
[x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
[x] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
```

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Documentation
[ ] Other... Please describe:
```

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/ADF-696

Entire group header is not clickable
Accordion (expand) icon hiding logic doesn't work properly.

**What is the new behaviour?**

Entire group header will be clickable
Added input show/hide accordion icon 

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.html [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <div class="adf-panel adf-panel-default" [ngClass]="{'adf-panel-open': isOpen}">
-    <div class="adf-panel-heading" [ngClass]="{'adf-panel-heading-selected': isSelected}">
-        <div (click)="onHeadingClick()">
+    <div class="adf-panel-heading" [ngClass]="{'adf-panel-heading-selected': isSelected}" (click)="onHeadingClick()">
+        <div>
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.html]: This div is not needed most probably anymore

File: ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.ts [hunk 2]
```diff
@@ -37,6 +37,9 @@ export class AccordionGroupComponent implements OnDestroy {
     @Input()
     headingIcon: string;
 
+    @Input()
+    hasAccordionIcon: boolean = true;
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.ts]: Documentation of this property is missing

File: ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.spec.ts [hunk 3]
```diff
@@ -75,25 +75,26 @@ describe('AccordionGroupComponent', () => {
         });
     });
 
-    it('should hide expand icon by default', () => {
+    it('should show expand icon by default', () => {
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.spec.ts]: Why this default behavior is changed ? I don't see this modify in the PR

File: ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.ts (hunk 2 above)
Comment (monsterflaw) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.ts]: Waiting for #1913 to be merged.

File: ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.spec.ts (hunk 3 above)
Comment (monsterflaw) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.spec.ts]: In the previous approach (https://github.com/Alfresco/alfresco-ng2-components/commit/22bbffdb2162cb3ed670c5797dad7a54e0ac3d27), accordion icon hiding logic was solely based on the length of the inner HTML and when we used this with ADF Dashboard, we faced issues when the child elements were conditionally added with ng-if directive. So we thought of having an input and let the developer decide when the expand icon needs to be hidden.

The new input (hasAccordionIcon) is optional and the default value is true, so default behaviour is showing the expand icon and the same is hidden only when the hasAccordionIcon input is explicitly set as false.

Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.spec.ts]: ok got it , thanks for the explanation

File: ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.ts (hunk 2 above)
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/components/collapsable/accordion-group.component.ts]: has been merged

**Pull request closed without merging**